pkg/proxy/healthcheck/proxier_health.go

```diff
@@ -80,12 +80,17 @@ func newProxierHealthServer(listener listener, httpServerFactory httpServerFacto
 
 // Updated updates the lastUpdated timestamp.
 func (hs *proxierHealthServer) Updated() {
+	hs.oldestPendingQueued.Store(time.Time{})
 	hs.lastUpdated.Store(hs.clock.Now())
 }
 
 // QueuedUpdate updates the lastQueued timestamp.
 func (hs *proxierHealthServer) QueuedUpdate() {
-	hs.lastQueued.Store(hs.clock.Now())
+	if val := hs.oldestPendingQueued.Load(); val != nil {
+		if val.(time.Time).IsZero() {
```

Comment:
The docs for `atomic.Value` say "The zero value for a Value returns nil from Load."
So this is wrong

Reply:
Nope,  I made some tests, If `val.(time.Time).IsZero()` , The return value is non nil from `Load`. See the following:
```golang
// QueuedUpdate updates the lastQueued timestamp.
func (hs *proxierHealthServer) QueuedUpdate() {
	if val := hs.oldestPendingQueued.Load(); val == nil {
		hs.oldestPendingQueued.Store(hs.clock.Now())
	} else {
		fmt.Println("val: ",val.(time.Time))
		if val.(time.Time).IsZero() {
			hs.oldestPendingQueued.Store(hs.clock.Now())
		}
	}
}
```

Result:
```bash
/Users/cyclinder/Downloads/goland/go1.18.1/bin/go tool test2json -t /private/var/folders/nl/k22q6s052q76dfr61mft5stm0000gn/T/GoLand/___TestHealthzServer_in_k8s_io_kubernetes_pkg_proxy_healthcheck.test -test.v -test.paniconexit0 -test.run ^\QTestHealthzServer\E$
=== RUN   TestHealthzServer
val:  0001-01-01 00:00:00 +0000 UTC
--- PASS: TestHealthzServer (0.00s)
PASS

Process finished with the exit code 0
```